apps/web/src/components/Solana/ConnectWalletModal.tsx

```diff
@@ -0,0 +1,169 @@
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { createPortal } from "react-dom";
+import { twMerge } from "tailwind-merge";
+import { useWalletModal } from "./useWalletModal";
+import { WalletListItem } from "./WalletListItem";
+
+export interface WalletModalProps {
+  className?: string;
+  container?: string;
+}
+
+export const WalletModal: FC<WalletModalProps> = ({ container = "body" }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const {
+    wallets,
+    select,
+    wallet: selectedWallet,
+    connecting,
+    connected,
+  } = useWallet();
+  const { setVisible } = useWalletModal();
+  const [fadeIn, setFadeIn] = useState(false);
+  const [portal, setPortal] = useState<Element | null>(null);
+
+  const [installedWallets, notInstalledWallets] = useMemo(() => {
+    const installed: Wallet[] = [];
+    const notInstalled: Wallet[] = [];
+
+    for (const wallet of wallets) {
+      if (wallet.readyState === WalletReadyState.Installed) {
+        installed.push(wallet);
+      } else {
+        notInstalled.push(wallet);
+      }
+    }
+
+    return [installed, notInstalled];
+  }, [wallets]);
+
+  const hideModal = useCallback(() => {
+    setFadeIn(false);
+    setTimeout(() => setVisible(false), 150);
+  }, [setVisible]);
+
+  const handleClose = useCallback(
+    (event: MouseEvent) => {
+      event.preventDefault();
+      hideModal();
+    },
+    [hideModal],
+  );
+
+  const handleWalletClick = useCallback(
+    (_event: MouseEvent, walletName: WalletName) => {
+      select(walletName);
+      // handleClose(event);
+    },
+    [select],
+  );
+
+  useEffect(() => {
+    if (selectedWallet && connected) {
+      hideModal();
+    }
+  }, [selectedWallet, connected, hideModal]);
+
+  useLayoutEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        hideModal();
+      }
+    };
+
+    // Get original overflow
+    const { overflow } = window.getComputedStyle(document.body);
+    // Hack to enable fade in animation after mount
+    setTimeout(() => setFadeIn(true), 0);
+    // Prevent scrolling on mount
+    document.body.style.overflow = "hidden";
+    // Listen for keydown events
+    window.addEventListener("keydown", handleKeyDown, false);
+
+    return () => {
+      // Re-enable scrolling when component unmounts
+      document.body.style.overflow = overflow;
+      window.removeEventListener("keydown", handleKeyDown, false);
+    };
+  }, [hideModal]);
+
+  useLayoutEffect(
+    () => setPortal(document.querySelector(container)),
+    [container],
+  );
+
+  useEffect(() => {
+    if (fadeIn && ref.current) {
+      ref.current.focus();
+    }
+  }, [fadeIn]);
+
+  return (
+    portal &&
+    createPortal(
+      <div
+        className={twMerge(
+          "fixed top-0 right-0 flex h-screen w-110 items-center justify-center bg-transparent p-5 transition-opacity duration-150",
+          fadeIn ? "opacity-100" : "opacity-0",
```

Comment:
We shouldn't construct class name dynamically.

https://tailwindcss.com/docs/detecting-classes-in-source-files#dynamic-class-names